core/src/test/kotlin/fr/sncf/osrd/stdcm/preprocessing/BlockAvailabilityTests.kt

```diff
@@ -636,4 +636,330 @@ class BlockAvailabilityTests {
             ) as BlockAvailabilityInterface.Unavailable
         assertEquals(120.0, res.duration)
     }
+
+    /** Train goes through planned step at the right moment: should return available. */
+    @Test
+    fun testPlannedStepRespected() {
+        val explorer = makeExplorer(5, 1)
+        val plannedStepOffset = Offset<Block>(0.meters)
+        val timeAtZoneEnd = explorer.interpolateDepartureFromClamp(explorer.getSimulatedLength())
+        val steps =
+            listOf(
+                STDCMStep(
+                    listOf(PathfindingEdgeLocationId(blocks[0], plannedStepOffset)),
+                    null,
+                    false,
+                    PlannedTimingData(0.seconds, 0.seconds, timeAtZoneEnd.seconds)
+                )
+            )
+        val availability = makeBlockAvailability(infra, listOf(), listOf(), steps)
+        val res =
+            availability.getAvailability(
+                explorer,
+                Offset(0.meters),
+                explorer.getSimulatedLength(),
+                0.0
+            ) as BlockAvailabilityInterface.Available
+        assertEquals(timeAtZoneEnd.seconds.seconds, res.maximumDelay)
+    }
+
+    /**
+     * Train goes through planned step and conflict at the right moment: should return available.
+     */
+    @ParameterizedTest
+    @CsvSource("30.0", "60.0")
+    fun testAvailablePlannedStepBeforeAvailableConflict(conflictMaximumDelay: Double) {
+        val explorer = makeExplorer(5, 1)
+
+        val plannedStepOffset = Offset<Block>(100.meters)
+        val timeAtStep = explorer.interpolateDepartureFromClamp(Offset(plannedStepOffset.distance))
+        val stepMaximumTolerance = 60.0
+        val steps =
+            listOf(
+                STDCMStep(
+                    listOf(PathfindingEdgeLocationId(blocks[0], plannedStepOffset)),
+                    null,
+                    false,
+                    PlannedTimingData(
+                        30.seconds,
+                        (stepMaximumTolerance / 2).seconds,
+                        (stepMaximumTolerance / 2).seconds
+                    )
+                )
+            )
+        val stepMaximumDelay = stepMaximumTolerance - timeAtStep
```

Comment:
The semantics are a little weird in this sequence.


`val stepMaximumDelay = stepMaximumTolerance - timeAtStep` means that the last available time is `stepMaximumTolerance`. It's true, but not 100% obvious.

We could introduce a variable named `lastAvailableTime` and define it as 30s + tolerance / 2. Which does result in 60s = `stepMaximumTolerance`, but it would explain how we got there.

Reply:
I kinda went another way for the following tests, and should have modified this one. Will make it clearer ;)